validator/client/propose.go

```diff
@@ -429,3 +446,30 @@ func (v *validator) getGraffiti(ctx context.Context, pubKey [fieldparams.BLSPubk
 
 	return []byte{}, nil
 }
+
+func (v *validator) SetGraffiti(ctx context.Context, pubKey [fieldparams.BLSPubkeyLength]byte, graffiti []byte) error {
+	if v.proposerSettings == nil || v.proposerSettings.ProposeConfig == nil {
+		return errors.New("attempted to set graffiti without proposer settings, graffiti will default to flag options")
+	}
+	ps := v.proposerSettings.Clone()
+	var option *validatorserviceconfig.ProposerOption
+	option, ok := ps.ProposeConfig[pubKey]
+	if !ok || option == nil {
+		return fmt.Errorf("attempted to set graffiti but proposer settings are missing for pubkey:%s", hexutil.Encode(pubKey[:]))
+	}
+	option.Graffiti = string(graffiti)
+	return v.SetProposerSettings(ctx, ps) // save the proposer settings
+}
+
+func (v *validator) DeleteGraffiti(ctx context.Context, pubKey [fieldparams.BLSPubkeyLength]byte) error {
+	if v.proposerSettings == nil || v.proposerSettings.ProposeConfig == nil {
+		return errors.New("attempted to delete graffiti without proposer settings, graffiti will default to flag options")
+	}
+	ps := v.proposerSettings.Clone()
+	option, ok := ps.ProposeConfig[pubKey]
+	if !ok || option == nil {
+		return fmt.Errorf("graffiti not found in proposer settings for pubkey:%s", hexutil.Encode(pubKey[:]))
+	}
+	option.Graffiti = ""
```

Comment:
In this case and "empty string" graffiti is the same than no graffiti at all.
A user may set an empty graffiti on purpose.

I guess a solution is to set `option.Graffiti` as a pointer to `Graffiti` and test is the pointer is `nil`.

Reply:
Example, if I use the following file:

```
default_config:
  fee_recipient: 0x1268AD189526AC0b386faF06eFfC46779c340eE6
  graffiti: Default Graffiti
proposer_config:
  0x80f408a41341416f79e3be605c5f1b960169bea5753b9b8181f7466b4fa0b13addb250c9d068af263f820c113536f9f7:
    fee_recipient: 0xef01d47ed04b181a2715D4050208f7d6e381bF42
  0xa17524bcf06d4fbf042e17dae68637368607f4fc9c28c104f49b12d1e0bed93fa3714f35ff05f4e518f789319c9b8f86:
    fee_recipient: 0x801f87A1e14398e2ec9031CaBd585A84D92BC606
  0x84d7fa31925c559ef85531687b190d9299036625a73370153502bf1c086ac7d7e3006c15210c6f9bd4b518c1900e0f45:
    fee_recipient: 0x801f87A1e14398e2ec9031CaBd585A84D92BC606
    graffiti: ""
```

Then querying the graffiti for `0x84...` will return `Default Graffiti`, while `graffiti: ""` has specifically been set by the user.